cli/azd/pkg/environment/storage_blob_data_store.go

```diff
@@ -143,6 +153,22 @@ func (sbd *StorageBlobDataStore) Save(ctx context.Context, env *Environment, opt
 		return fmt.Errorf("uploading .env: %w", describeError(err))
 	}
 
+	// Upload Infra Files if any
+	filesToUpload := sbd.azdContext.GetEnvironmentInfraFiles(env.name)
+
+	for _, file := range filesToUpload {
+		fileName := filepath.Base(file)
+		fileBuffer, err := os.Open(file)
+		if err != nil {
+			return fmt.Errorf("failed to open file for upload: %w", err)
+		}
+		defer fileBuffer.Close()
+		err = sbd.blobClient.Upload(ctx, fmt.Sprintf("%s/%s/%s", env.name, InfraFilesDir, fileName), fileBuffer)
```

Comment:
How are secrets protected in this approach? Seems like a user could upload a lot of sensitive information to a storage account that I don't see any documentation on how to secure correctly.  There's no mention of turning on blob encryption, turning off public access, restricting network access, or any other best practices for putting tfstate in a storage account https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/remote-environments-support

If I'm working across two computers, how is the other computer notified of the change so I have the latest state?  Do I have to manually re-run `azd env refresh`?